src/platform/unix/mod.rs

```diff
@@ -953,21 +976,32 @@ fn is_socket(fd: c_int) -> bool {
 
 // FFI stuff follows:
 
-const SCM_RIGHTS: c_int = 0x01;
+#[allow(non_snake_case)]
+fn CMSG_LEN(length: SockLen) -> SockLen {
+    (CMSG_ALIGN(mem::size_of::<cmsghdr>()) + length) as SockLen
+}
+
+#[allow(non_snake_case)]
+fn CMSG_ALIGN(length: size_t) -> SockLen {
+    ((length + mem::size_of::<size_t>() - 1) & !(mem::size_of::<size_t>() - 1)) as SockLen
+}
 
 #[allow(non_snake_case)]
-fn CMSG_LEN(length: size_t) -> size_t {
-    CMSG_ALIGN(mem::size_of::<cmsghdr>()) + length
+#[cfg(target_os = "linux")]
+unsafe fn CMSG_DATA(cmsg: *mut cmsghdr) -> *mut c_void {
+    cmsg.offset(1) as *mut c_void
 }
 
 #[allow(non_snake_case)]
-fn CMSG_ALIGN(length: size_t) -> size_t {
-    (length + mem::size_of::<size_t>() - 1) & !(mem::size_of::<size_t>() - 1)
+#[cfg(target_os = "freebsd")]
+unsafe fn CMSG_DATA(cmsg: *mut cmsghdr) -> *mut c_void {
```

Comment:
Ensure proper alignment. `cmsg.offset(1)` on freebsd is the equivalent of `addq 0xc, %rcx`, but it should be `addq 0x10, %rcx`.